backend typescript build fails

**Describe the bug**
During typescript build, the following errors appear:

```
src/filters/global-exception.filter.ts:15:27 - error TS2339: Property 'message' does not exist on type '{}'.
15       message: exception?.message || 'Unknown error',
                             ~~~~~~~
src/filters/global-exception.filter.ts:16:25 - error TS2339: Property 'stack' does not exist on type '{}'.
16       stack: exception?.stack,
                           ~~~~~
src/filters/global-exception.filter.ts:53:24 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type '"uncaughtException"' is not assignable to parameter of type 'Signals'.
53           process.emit('uncaughtException', exception);
                          ~~~~~~~~~~~~~~~~~~~
  node_modules/@types/node/process.d.ts:1834:17
    1834                 emit(event: Signals, signal?: Signals): boolean;
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.
src/filters/global-exception.filter.ts:59:20 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type '"uncaughtException"' is not assignable to parameter of type 'Signals'.
59       process.emit('uncaughtException', exception);
                      ~~~~~~~~~~~~~~~~~~~
  node_modules/@types/node/process.d.ts:1834:17
    1834                 emit(event: Signals, signal?: Signals): boolean;
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.
       Found 4 error(s).
```


Use `nest build` to reproduce.

--- a/teammapper-backend/src/filters/global-exception.filter.ts
+++ b/teammapper-backend/src/filters/global-exception.filter.ts
@@ -6,16 +6,18 @@ import { ExceptionFilter, Catch, ArgumentsHost, Logger, HttpException } from '@n
 export class GlobalExceptionFilter implements ExceptionFilter {
   private readonly logger = new Logger(GlobalExceptionFilter.name);
 
-  catch(exception: Error | HttpException | unknown, host: ArgumentsHost) {
+  catch(exception: unknown, host: ArgumentsHost) {
     const ctx = host.getType();
 
-    this.logger.error({
+    const errorDetails = {
       error: exception,
       type: exception?.constructor?.name || typeof exception,
-      message: exception?.message || 'Unknown error',
-      stack: exception?.stack,
+      message: exception instanceof Error ? exception.message : 'Unknown error',
+      stack: exception instanceof Error ? exception.stack : undefined,
       context: ctx,
-    });
+    };
+
+    this.logger.error(errorDetails);
 
     try {
       switch (ctx) {
@@ -49,14 +51,28 @@ export class GlobalExceptionFilter implements ExceptionFilter {
         default: {
           // Handle any runtime errors outside HTTP/WS contexts
           this.logger.error(`Unhandled exception type: ${ctx}`);
-          // Emit to process handler as last resort
-          process.emit('uncaughtException', exception);
+          // Forward to the global error handler
+          if (exception instanceof Error) {
+            process.emitWarning(exception);
+          } else {
+            process.emitWarning(
+              new Error(String(exception)),
+              'UnhandledError'
+            );
+          }
         }
       }
     } catch (handlerError) {
       // If the error handler itself fails, log it and emit to process
       this.logger.error('Global exception handler failed: ', handlerError);
-      process.emit('uncaughtException', exception);
+      if (handlerError instanceof Error) {
+        process.emitWarning(handlerError);
+      } else {
+        process.emitWarning(
+          new Error(String(handlerError)),
+          'HandlerError'
+        );
+      }
     }
   }
 }

--- a/teammapper-backend/src/filters/global-exception.filter.ts
+++ b/teammapper-backend/src/filters/global-exception.filter.ts
@@ -1,13 +1,18 @@
-import { ExceptionFilter, Catch, ArgumentsHost, Logger, HttpException } from '@nestjs/common';
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  Logger
+} from '@nestjs/common'
 
 // This is for any unhandled gateway and "internal" NestJS related errors - like if the gateway can't reach clients or things like that.
 // It will try to always keep clients and their websockets alive and gracefully send errors over the wire, without revealing internal error reasons.
 @Catch()
 export class GlobalExceptionFilter implements ExceptionFilter {
-  private readonly logger = new Logger(GlobalExceptionFilter.name);
+  private readonly logger = new Logger(GlobalExceptionFilter.name)
 
   catch(exception: unknown, host: ArgumentsHost) {
-    const ctx = host.getType();
+    const ctx = host.getType()
 
     const errorDetails = {
       error: exception,
@@ -15,63 +20,57 @@ export class GlobalExceptionFilter implements ExceptionFilter {
       message: exception instanceof Error ? exception.message : 'Unknown error',
       stack: exception instanceof Error ? exception.stack : undefined,
       context: ctx,
-    };
+    }
 
-    this.logger.error(errorDetails);
+    this.logger.error(errorDetails)
 
     try {
       switch (ctx) {
         case 'http': {
-          const response = host.switchToHttp().getResponse();
+          const response = host.switchToHttp().getResponse()
           return response.status(500).json({
             statusCode: 500,
             message: 'Internal server error',
             timestamp: new Date().toISOString(),
-          });
+          })
         }
 
         case 'ws': {
-          const client = host.switchToWs().getClient();
+          const client = host.switchToWs().getClient()
           const error = {
             event: 'error',
             data: {
               message: 'Internal server error',
               timestamp: new Date().toISOString(),
             },
-          };
+          }
 
           if (typeof client.emit === 'function') {
-            client.emit('error', error);
+            client.emit('error', error)
           } else if (typeof client.send === 'function') {
-            client.send(JSON.stringify(error));
+            client.send(JSON.stringify(error))
           }
-          break;
+          break
         }
 
         default: {
           // Handle any runtime errors outside HTTP/WS contexts
-          this.logger.error(`Unhandled exception type: ${ctx}`);
+          this.logger.error(`Unhandled exception type: ${ctx}`)
           // Forward to the global error handler
           if (exception instanceof Error) {
-            process.emitWarning(exception);
+            process.emitWarning(exception)
           } else {
-            process.emitWarning(
-              new Error(String(exception)),
-              'UnhandledError'
-            );
+            process.emitWarning(new Error(String(exception)), 'UnhandledError')
           }
         }
       }
     } catch (handlerError) {
       // If the error handler itself fails, log it and emit to process
-      this.logger.error('Global exception handler failed: ', handlerError);
+      this.logger.error('Global exception handler failed: ', handlerError)
       if (handlerError instanceof Error) {
-        process.emitWarning(handlerError);
+        process.emitWarning(handlerError)
       } else {
-        process.emitWarning(
-          new Error(String(handlerError)),
-          'HandlerError'
-        );
+        process.emitWarning(new Error(String(handlerError)), 'HandlerError')
       }
     }
   }

--- a/teammapper-backend/src/main.ts
+++ b/teammapper-backend/src/main.ts
@@ -10,8 +10,8 @@ async function bootstrap() {
 
   // Process-level handlers for uncaught errors - anything that happens outside of NestJS, such as type errors.
   // This is only logged server-side so we log the whole stack for better review.
-  process.on('uncaughtException', (error) => {
-    logger.error('Uncaught Exception: ', error.stack);
+  process.on('warning', (error) => {
+    logger.error('Possible uncaught exception: ', error);
   });
 
   process.on('unhandledRejection', (reason: unknown) => {

--- a/teammapper-backend/src/main.ts
+++ b/teammapper-backend/src/main.ts
@@ -2,28 +2,29 @@ import { NestFactory } from '@nestjs/core'
 import AppModule from './app.module'
 import configService from './config.service'
 import { createProxyMiddleware } from 'http-proxy-middleware'
-import { GlobalExceptionFilter } from './filters/global-exception.filter';
+import { GlobalExceptionFilter } from './filters/global-exception.filter'
 import { Logger } from '@nestjs/common'
 
 async function bootstrap() {
-  const logger = new Logger('Main Process');
+  const logger = new Logger('Main Process')
 
   // Process-level handlers for uncaught errors - anything that happens outside of NestJS, such as type errors.
   // This is only logged server-side so we log the whole stack for better review.
   process.on('warning', (error) => {
-    logger.error('Possible uncaught exception: ', error);
-  });
+    logger.error('Possible uncaught exception: ', error)
+  })
 
   process.on('unhandledRejection', (reason: unknown) => {
-    const stack = reason instanceof Error ? reason.stack : 'No stack trace available';
-    logger.error('Unhandled Rejection. Stack trace: ', stack);
-  });
+    const stack =
+      reason instanceof Error ? reason.stack : 'No stack trace available'
+    logger.error('Unhandled Rejection. Stack trace: ', stack)
+  })
 
   const app = await NestFactory.create(AppModule, {
     logger: ['log', 'error', 'warn', 'debug'],
   })
 
-  app.useGlobalFilters(new GlobalExceptionFilter());
+  app.useGlobalFilters(new GlobalExceptionFilter())
 
   app.use(
     '/arasaac/api',

